src/unix/darwin.c

```diff
@@ -333,3 +333,29 @@ void uv_free_interface_addresses(uv_interface_address_t* addresses,
 
   uv__free(addresses);
 }
+
+int uv__tcp_enable_largesocket(uv_tcp_t *tcp) {
+  int rmem;
+  int wmem;
+  int r;
+
+  size_t size = sizeof(int);
+  if (sysctlbyname("net.inet.tcp.recvspace", &rmem, &size, NULL, 0) ||
+      rmem <= STDTCPWINDOW)
+    return UV_ENOSYS;
+  
+  if (sysctlbyname("net.inet.tcp.sendspace", &wmem, &size, NULL, 0) ||
```

Comment:
I think this implementation could probably work on the BSD's (FreeBSD, OpenBSD, etc). If that's the case this could be moved to a separate file so could be used by every platform.